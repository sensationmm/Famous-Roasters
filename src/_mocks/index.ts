import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
