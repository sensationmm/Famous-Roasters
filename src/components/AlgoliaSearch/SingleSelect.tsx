import { Listbox as HUIListbox, Transition } from '@headlessui/react'
import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/outline'
import { Fragment } from 'react'

import { Typography, TypographySize, TypographyType } from '../Typography'

export interface SelectItem {
  label: string
  value: string
}

interface SingleSelectProps {
  items: SelectItem[]
  value: SelectItem
  onChange: (value: SelectItem) => void
  big?: boolean
}

const SingleSelect = ({ items, value, onChange, big }: SingleSelectProps) => {
  const renderButton = (open: boolean) => {
    const renderIcon = () => {
      if (big) {
        if (open) {
          return <ChevronUpIcon className="-mr-1 ml-2 h-6 w-6 self-center" aria-hidden="true" />
        }
        return <ChevronDownIcon className="-mr-1 ml-2 h-6 w-6 self-center" aria-hidden="true" />
      } else {
        if (open) {
          return <ChevronUpIcon className="-mr-1 ml-2 h-6 w-6" aria-hidden="true" />
        }
        return <ChevronDownIcon className="-mr-1 ml-2 h-6 w-6" aria-hidden="true" />
      }
    }

    const classNames = ['block', 'truncate']

    return big ? (
      <HUIListbox.Button
        className="inline-flex justify-between py-2 pr-1 border-b-2 border-brand-black text-left bg-white cursor-default"
        data-testid="button-listbox"
      >
        <Typography type={TypographyType.Heading} size={TypographySize.Tiny} className={classNames.join(' ')}>
          {value.label}
        </Typography>
        {renderIcon()}
      </HUIListbox.Button>
    ) : (
      <HUIListbox.Button
        className={`inline-flex justify-between w-full text-left bg-white rounded-full border border-coreUI-text-tertiary cursor-default px-4 py-2`}
        data-testid="button-listbox"
      >
        <Typography size={TypographySize.Base} className={`${classNames.join(' ')}`}>
          {value.label}
        </Typography>
        {renderIcon()}
      </HUIListbox.Button>
    )
  }

  return (
    <div className={'w-auto min-w-fit max-w-xs relative'}>
      <HUIListbox value={value} onChange={onChange}>
        {({ open }) => (
          <>
            <div className="grid grid-cols-2 items-end"></div>
            {renderButton(open)}
            <Transition
              as={Fragment}
              leave="transition ease-in duration-100"
              leaveFrom="opacity-100"
              leaveTo="opacity-0"
            >
              <HUIListbox.Options className="absolute z-10 min-w-full mt-1 overflow-auto text-base bg-white rounded-md shadow-lg max-h-60 border border-coreUI-background-images focus:outline-none">
                {items.map((option, idx) => (
                  <HUIListbox.Option
                    key={option.value}
                    data-testid={`option-${idx}`}
                    className={({ active }) =>
                      `cursor-pointer select-none relative py-2 pl-8 pr-4 ${
                        active ? 'bg-brand-grey-whisper' : 'bg-transparent'
                      }`
                    }
                    value={option}
                  >
                    {() => {
                      const isSelected = option.value === value.value
                      return (
                        <>
                          <span
                            className={`block truncate flex align-middle ${
                              isSelected ? 'font-semibold' : 'font-normal'
                            } `}
                          >
                            {option.label}
                          </span>
                          {isSelected ? (
                            <span className="absolute inset-y-0 left-0 flex items-center pl-1.5">
                              <CheckIcon className="w-5 h-5 text-brand-green-club" aria-hidden="true" />
                            </span>
                          ) : null}
                        </>
                      )
                    }}
                  </HUIListbox.Option>
                ))}
              </HUIListbox.Options>
            </Transition>
          </>
        )}
      </HUIListbox>
    </div>
  )
}

export default SingleSelect
