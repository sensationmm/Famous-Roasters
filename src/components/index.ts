import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
