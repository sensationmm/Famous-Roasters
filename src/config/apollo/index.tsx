import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
