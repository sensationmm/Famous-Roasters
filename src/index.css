@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}