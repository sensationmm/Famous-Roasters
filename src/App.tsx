import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
