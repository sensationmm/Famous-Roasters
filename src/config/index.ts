import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
