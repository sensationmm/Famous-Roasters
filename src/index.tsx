import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
