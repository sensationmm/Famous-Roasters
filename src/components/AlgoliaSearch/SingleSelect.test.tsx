import { render, waitFor } from '@testing-library/react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter as Router } from 'react-router-dom'
import { i18n } from 'src/config'

import SingleSelect from './SingleSelect'

const items = [
  { label: 'Alle Kaffees', value: '' },
  { label: 'Espresso', value: 'espresso' },
  { label: 'Filter', value: 'filter' },
  { label: 'Omni', value: 'omni' },
]

describe('SingleSelect component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <Router>
        <I18nextProvider i18n={i18n}>
          <SingleSelect items={items} value={items[0]} onChange={jest.fn()} />
        </I18nextProvider>
      </Router>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
