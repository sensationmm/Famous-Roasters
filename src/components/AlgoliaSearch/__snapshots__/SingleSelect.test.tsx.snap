// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSelect component Renders correctly 1`] = `
<div>
  <div
    class="w-auto min-w-fit max-w-xs relative"
  >
    <div
      class="grid grid-cols-2 items-end"
    />
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="inline-flex justify-between w-full text-left bg-white rounded-full border border-coreUI-text-tertiary cursor-default px-4 py-2"
      data-headlessui-state=""
      data-testid="button-listbox"
      id="headlessui-listbox-button-:r0:"
      type="button"
    >
      <span
        class="block truncate text-base leading-6"
      >
        Alle Kaffees
      </span>
      <svg
        aria-hidden="true"
        class="-mr-1 ml-2 h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 9l-7 7-7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
